### PR TITLE
Update bindings to 2.12

### PR DESCRIPTION
**Pull request by teo-tsirpanis**

### CODE REVIEW

File: sources/TileDB.CSharp/Enums.cs [hunk 1]
```diff
@@ -18,11 +18,13 @@ public enum ObjectType : uint
         TILEDB_ARRAY = tiledb_object_t.TILEDB_ARRAY
     }
 
-
     public enum QueryType : uint
     {
         TILEDB_READ = tiledb_query_type_t.TILEDB_READ,
         TILEDB_WRITE = tiledb_query_type_t.TILEDB_WRITE,
+        Delete = tiledb_query_type_t.TILEDB_DELETE,
```
Comment (teo-tsirpanis) [inline sources/TileDB.CSharp/Enums.cs:25]: I gave C#-friendly names to the new enum values, will update the others soon (and obsolete the `TILEDB_***` values).

**Review by shaunrd0 — CHANGES_REQUESTED** [review]
Looks good! I like the changes to `generate.rsp`, these options will make the regeneration commands much simpler. :+1: I regenerated locally and had the same results.

We should also update `.csproj` and `.nuspec` files, this is something I forgot to do on my 2.12 branch so [here](https://github.com/TileDB-Inc/TileDB-CSharp/pull/100/files#diff-1d244d9aab29d2ed08444d57b10ceb3ed925e3bb990287d6ea94f5097a13fb55) is an example of the changes I made to these files before rolling out a new release for 2.11.3.

File: cpp/cmake/TileDB.cmake [hunk 2]
```diff
@@ -416,7 +427,7 @@ elseif(${TILEDB_VERSION} STREQUAL "2.0.7")
     SET(TILEDB_DOWNLOAD_URL "https://github.com/TileDB-Inc/TileDB/releases/download/2.0.7/tiledb-linux-2.0.7-2058d3d.tar.gz")
     SET(TILEDB_DOWNLOAD_SHA1 "556CCD6265A9F62884A8F798753625D7352ACCD5")
   endif()
-else()
+elseif(${TILEDB_VERSION} STREQUAL "2.11.3")
```
[the remark this answers is not in the document]
Comment (teo-tsirpanis) [reply cpp/cmake/TileDB.cmake]: Done.

**Comment by teo-tsirpanis** [comment]
I updated the library's version to 5.3.0 and removed it from the example projects; they don't need one.

**Pull request merged**